File: apps/backend/app/schemas/point_purchase.py
# app/schemas/point_purchase.py
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, Field, ConfigDict
from typing import List, Optional
from .point_plan import PointPlanRead
from enum import Enum

class PurchaseStatus(str, Enum):
    PENDING   = "PENDING"
    PAID      = "PAID"
    FAILED    = "FAILED"
    CANCELLED = "CANCELLED"

class PointPurchaseCreate(BaseModel):
    plan_id: UUID

class PointPurchaseRead(BaseModel):
    id: UUID
    plan: Optional[PointPlanRead]
    amount: float
    status: PurchaseStatus
    asaas_id: str
    pix_qr_code: Optional[str]
    pix_copy_paste_code: Optional[str]
    pix_expires_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

class PaginatedPointPurchases(BaseModel):
    total: int
    skip: int
    limit: int
    items: List[PointPurchaseRead]

File: apps/portal-client/src/types/pointPurchase.ts
// src/types/pointPurchase.ts
import type { CompanyBasic } from '@/types/company';
/** Status de uma compra de pontos */
export type PurchaseStatus = 'PENDING' | 'PAID' | 'FAILED' | 'CANCELLED';

/** Leitura de uma compra de pontos */
export interface PointPurchaseRead {
  id: string;                  // UUID
  plan?: PointPlanRead;        // pode vir null
  amount: number;
  status: PurchaseStatus;
  asaas_id: string;
  company: CompanyBasic
  pix_qr_code?: string;
  pix_copy_paste_code?: string;
  pix_expires_at?: string;     // ISO datetime
  created_at: string;          // ISO datetime
  updated_at: string;          // ISO datetime
}

/** Páginação de compras de pontos */
export interface PaginatedPointPurchases {
  total: number;
  skip: number;
  limit: number;
  items: PointPurchaseRead[];
}

// note: PointPlanRead é importado do seu types/pointPlan.ts
import type { PointPlanRead } from './pointPlan';
